app/scripts/ui.js

```diff
@@ -37,6 +37,17 @@ async function start() {
   const activeTab = await queryCurrentActiveTab(windowType);
   initializeUiWithTab(activeTab);
 
+  if (process.env.ENABLE_MV3) {
+    extensionPort.onMessage.addListener((message) => {
+      if (
+        document.getElementById('app-loader') &&
+        message?.name === 'CONNECTION_READY'
+      ) {
+        initializeUiWithTab(activeTab);
+      }
+    });
+  }
+
```

Comment:
I have added this message above to initialise UI when backend is ready, otherwise in if UI initialises before backend is ready it results in blank screen in extension and this problem was very frequent with MV3.

Reply:
calling `initializeUiWithTab` twice in this case, is that intentional?